### PR TITLE
Masthead vanilla sb

### LINKED ISSUE #1811: Add Masthead Component - Vanilla NDS
**User Story:**
As Spark we want to add the component to Storybook so that it will render on the new doc site.

**Notes:**
- Actions & Knobs will follow up in future Issues

**AC:**
- Add Component Code will render on the page

### PR DESCRIPTION
**Pull request by RVMendoza**
## What does this PR do?
Adds masthead vanilla to storybook
image
image

### Associated Issue 
Fixes #1811 

### Code
 - [x] Build Component in Spark Vanilla (Sass, HTML, JS)
 - [x] Unit Testing in Spark Vanilla with `npm run test` (100% coverage, 100% passing)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [x] Apple Safari (Mobile)

### CODE REVIEW

File: spark/components/masthead/vanilla/masthead.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
 /* global document window */
+import { throttle } from 'lodash';
```
Comment (afebbraro) [inline spark/components/masthead/vanilla/masthead.js]: hey @RVMendoza , are these changes ones to bring the masthead.js file up to date?

File: spark/components/masthead/vanilla/masthead.stories.js [hunk 2]
```diff
@@ -1,3 +1,703 @@
...
+    toggle();
+    masthead();
+
+    // Our masthead.js applies load event to `window` but doesn't work in storybook iframes
+    // This useEffect runs this code when component mounts
+    const isMenuVisibleSB = isElementVisible('.sprk-c-Masthead__menu');
+    toggleScrollEvent(isMenuVisibleSB);
+  }, []);
+  return `
+    <header class="sprk-c-Masthead sprk-o-Stack" role="banner" idstring="masthead-2" data-sprk-masthead="">
```
Comment (afebbraro) [inline spark/components/masthead/vanilla/masthead.stories.js]:
```suggestion
    <header class="sprk-c-Masthead sprk-o-Stack" role="banner" idstring="masthead-2" data-sprk-masthead>
```
Our code doesn't need this to be set to an empty string. Just data-sprk-masthead is all we need.

File: spark/components/masthead/vanilla/masthead.stories.js [hunk 3]
```diff
@@ -1,3 +1,703 @@
...
+                </ul>
+              </div>
+            </li>
+          </ul>
+        </nav>
+      </div>
+
+      <div class="sprk-o-Stack__item">
+        <nav class="sprk-c-Masthead__big-nav" role="navigation">
+      <ul class="sprk-c-Masthead__big-nav-items sprk-o-Stack sprk-o-Stack--misc-a sprk-o-Stack--center-row sprk-o-Stack--split@xxs sprk-b-List sprk-b-List--bare" data-sprk-navigation="big">
```
Comment (afebbraro) [inline spark/components/masthead/vanilla/masthead.stories.js]: Some code alignment could be cleaned up here.

File: spark/components/masthead/vanilla/masthead.stories.js [hunk 4]
```diff
@@ -1,3 +1,703 @@
...
+        </li>
+
+        <li class="sprk-c-Masthead__big-nav-item sprk-o-Stack__item">
+          <a class="sprk-b-Link sprk-b-Link--plain sprk-c-Masthead__link sprk-c-Masthead__link--big-nav" href="#nogo">
+            Item
+          </a>
+        </li>
+      </ul>
+    </nav>
+
```
Comment (afebbraro) [inline spark/components/masthead/vanilla/masthead.stories.js]: Some extra empty lines could be cleaned up here.

File: spark/components/masthead/vanilla/masthead.stories.js [hunk 5]
```diff
@@ -1,3 +1,703 @@
...
+      </div>
+
+      <nav class="sprk-c-Masthead__narrow-nav sprk-u-Display--none" data-sprk-mobile-nav="mobileNav2" role="navigation" data-id="navigation-narrow-2">
+
+        <div data-sprk-masthead-mask="">
+          <div class="sprk-o-Box">
+            <a class="sprk-c-Masthead__selector sprk-b-Link sprk-b-Link--plain sprk-o-Stack sprk-o-Stack--split@xxs sprk-o-Stack--center-column" href="#nogo" data-sprk-dropdown-trigger="dropdown-selector" aria-haspopup="true">
```
Comment (afebbraro) [inline spark/components/masthead/vanilla/masthead.stories.js]: Some of the line lengths are beyond our maximum.

File: spark/components/masthead/vanilla/masthead.js [hunk 6]
```diff
@@ -1,7 +1,10 @@
...
 import getElements from '../../../utilities/helpers/vanilla/getElements';
 import { focusFirstEl } from '../../../utilities/helpers/vanilla/elementState';
 import { isEscPressed } from '../../../utilities/helpers/vanilla/keypress';
+import isElementVisible from '../../../utilities/helpers/vanilla/isElementVisible'
```
Comment (afebbraro) [inline spark/components/masthead/vanilla/masthead.js]:
```suggestion
import isElementVisible from '../../../utilities/helpers/vanilla/isElementVisible;'
```
Missing some semi-colons here.

File: spark/components/masthead/vanilla/masthead.js [hunk 7]
```diff
@@ -1,7 +1,10 @@
...
+import scrollYDirection from '../../../utilities/helpers/vanilla/scrollYDirection'
```
Comment (afebbraro) [inline spark/components/masthead/vanilla/masthead.js]: Missing a semi-colons here.

File: spark/components/masthead/vanilla/masthead.stories.js [hunk 8]
```diff
@@ -1,3 +1,703 @@
...
...3461-11.0631791-10.0653924 4.4812877c-1.5054326.6651911-3.1684104.8927565-4.7788732.6651911z" fill="#ff721e"/><path d="m31.859155 5.12897384 5.7066398-2.60824949c.4726358-.21006037.7352112-.70020121.6826961-1.20784709-.052515-.50764587-.4376257-.91026157-.9452716-.99778671-.5076458-.10503019-1.0152917-.15754528-1.5229376-.15754528h-.017505-1.2778673l-2.9758551 5.11146881z" fill="#ff721e"/><path d="m10.8881288 24.2619718c-.3676057-.8227364-.6126761-1.6979879-.7177062-2.6082495l-6.44185117 11.0806841 5.14647889 8.8575453c.24507042.4201207.73521125.6476861 1.22535208.542656l7.527163-12.9362174c-2.9408451-.3851106-5.4790745-2.2406438-6.7394366-4.9364185z" fill="#ff8920"/><path d="m15.3869216 12.6561368 16.1396377-7.36961768 2.9583501-5.11146881-11.8684104.03501005z" fill="#ff8920"/><path d="m10.6605634 17.5925554c.8052314-2.1706238 2.4682092-3.9211268 4.5863179-4.8663985l.1400403-.0525151 7.2295774-12.4460764h-5.5841046c-2.7657948.01750502-5.3215292 1.48792756-6.7044266 3.88611669l-9.31267605 16.24466801c-1.36539235 2.3981892-1.34788732 5.3565392.03501006 7.7372234l2.67826962 4.6213279 6.44185117-11.0806841c-.17505037-1.3653923 0-2.7482897.4901408-4.0436619z" fill="#ffa022"/></g></svg>
```
Comment (afebbraro) [inline spark/components/masthead/vanilla/masthead.stories.js]: Hey @RVMendoza this line exceeds our maximum line length. Can you shorten it up?

File: spark/components/masthead/vanilla/masthead.stories.js [hunk 9]
```diff
@@ -1,3 +1,703 @@
...
+            >
+                Sign In
+              </a>
+          </li>
+        </ul>
+      </nav>
+    </header>
+    <div class="sprk-u-pam">
+      <p class="sprk-u-mbm">
+        Molestie convallis gravida neque fermentum tellus vivamus praesent nisl curae; urna maecenas. Feugiat tortor pharetra vivamus scelerisque netus consectetur inceptos diam sociosqu fusce ac sem. Vestibulum congue mauris, facilisi aenean nulla. Erat habitant ipsum penatibus elit vulputate maecenas cras donec nisl leo. Accumsan sit scelerisque penatibus quam nisl. Torquent non nibh sollicitudin non. Sollicitudin rutrum gravida auctor a?
```
Comment (afebbraro) [inline spark/components/masthead/vanilla/masthead.stories.js]: Line lengths here.

File: spark/components/masthead/vanilla/masthead.js (hunk 1 above)
Comment (RVMendoza) [reply spark/components/masthead/vanilla/masthead.js]: yup!

**Pull request merged**